neo/Network/P2P/ProtocolHandler.cs

```diff
@@ -254,7 +254,8 @@ private void OnInventoryReceived(IInventory inventory)
         private void OnInvMessageReceived(InvPayload payload)
         {
             UInt256[] hashes = payload.Hashes.Where(p => knownHashes.Add(p)).ToArray();
```

Comment:
```suggestion
            UInt256[] hashes = payload.Hashes.Where(p => knownHashes.Add(p) && !sentHashes.HasItem(p)).ToArray();
```

Reply:
This suggested changes are useful but only for one line hahah

Reply:
I think it will be less efficient, no? But maybe it will be the same or better...ajajaj
That `Any` looks like something quite fast check...aheuaheauea

Reply:
Currently you look into the array twice, if `hashes` are empty, you try to find again in payload Hashes. i think that is faster all in the same query

Reply:
makes sense, @shargon.

Go for it. Do you want to apply the change? We need to change the `if` as well.

Reply:
Done